lua/bufferline/render.lua

```diff
@@ -113,43 +114,52 @@ local function render(update_names)
     local mod = is_modified and 'Mod' or ''
 
     local separatorPrefix = hl('Buffer' .. status .. 'Sign')
-    local separator = status == 'Inactive' and
-      icons.bufferline_separator_inactive or
-      icons.bufferline_separator_active
+    local separator = is_inactive and
+      icons.separator_inactive or
+      icons.separator_active
 
     local namePrefix = hl('Buffer' .. status .. mod)
     local name =
       (not has_icons and state.is_picking_buffer) and
         slice(buffer_name, 1) or
         buffer_name
 
-    local iconPrefix = ''
-    local icon = ''
+    local buffer_icons = {}
     if state.is_picking_buffer then
       local letter = JumpMode.get_letter(buffer_number)
-      iconPrefix = hl('Buffer' .. status .. 'Target')
-      icon =
-        (letter ~= nil and letter or ' ') ..
-        (has_icons and ' ' or '')
-    elseif has_icons then
-      if has_numbers then
-        local number_text = tostring(i)
-        iconPrefix = ''
-        icon = number_text .. (#number_text > 1 and '' or ' ')
-      else
-        local iconChar, iconHl = get_icon(buffer_name, vim.fn.getbufvar(buffer_number, '&filetype'), status)
-        iconPrefix = hl(status ~= 'Inactive' and iconHl or 'BufferInactive')
-        icon = iconChar .. ' '
-      end
+      buffer_icons[#buffer_icons+1] = {
+        -- Prefix
+        hl('Buffer' .. status .. 'Target'),
+        -- Icon
+        letter and letter..' ' or '',
+      }
+    elseif has_numbers then
+      local number_text = tostring(i)
+      buffer_icons[#buffer_icons+1] = {
+        -- Prefix
+        '',
+        -- Icon
+        number_text .. ' '
+      }
+    end
+
+    if has_icons then
+      local iconChar, iconHl = get_icon(buffer_name, vim.fn.getbufvar(buffer_number, '&filetype'), status)
+      buffer_icons[#buffer_icons+1] = {
+        --[[Prefix]]
+        hl(is_inactive and 'BufferInactive' or iconHl),
+        -- Icon
+        iconChar .. ' '
+      }
```

Comment:
As mentioned previously, I'd rather preserve the behavior that jump-mode doesn't introduce a new group but replaces an existing one. I like having it replace the icon. Also the current behavior is that it replace the first letter of the buffer name if there are no icons (to avoid layout shifts) but it doesn't seem to be taken into account in this PR.